controllers/post.controller.js

```diff
@@ -1,12 +1,39 @@
 import Post from '../models/post.model.js';
+import path from 'path';
+import { fileURLToPath } from 'url';
 import { fMsg } from "../utils/libby.js";
+import {deleteFile} from "../utils/libby.js";
 
-
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
 
 export const createPost = async (req, res) => {
     try {
+        let {
+            heading,
+            body,
+            contentPicture,
+            contentType,
+            reactions,
+            classname,
+            grade
+        } = req.body
+
+        const posted_by = req.user._id;
+        const school = req.user.schools[0] || "Some School";
+
+        contentPicture = req.file ? `/uploads/post_images/${req.file.filename}` : null;
+
         const post = new Post({
-            ...req.body,
+            posted_by,
+            school,
+            heading,
+            body,
+            contentPicture,
+            contentType,
+            reactions,
+            classname,
+            grade
         })
         await post.save();
         await post.populate('posted_by', 'userName email phone roles');
```

Comment:
I think we do not need to give the information like email and phone for the posts? we may need contentPicture.